### PR TITLE
[feat] Add support for repeated testing through two new options: `--reruns` and `--duration`

### LINKED ISSUE #619: Keeping a test system busy, new execution policy?
I would like to extend reframe for the use case of generating a constant workload on a test/ warm spare system. 

```
./reframe -C my_config.py --tag workload1 -r  --exec-policy=continuous --queue-depth=10
```

Where the main loop would be something like:

```python
while queue_depth < arg.queue_depth:
    start(random.choice(tests))
```

with the desired behavior to always maintain a queue of queue_depth pending jobs for the scheduler to evaluate.

I am willing to do the implementation myself, but would like some feedback on the reception of this idea and possibly some suggestions on how to implement it. e.g. Should it be a new execution policy? If so what should it be called?

### PR DESCRIPTION
**Pull request by vkarak**
This PR introduces the following two options that are meant to allow users to use reframe for stress testing a target system:

- The `--reruns=N` option will rerun the same test suite N times (N+1 total runs).
- The `--duration=TIMEOUT` option will run the same test suite repeatedly until `TIMEOUT` expires.

Compared to the similar `--max-retries` option that reruns the failed tests, these options will report statistics as well as failures from _all_ runs.

Closes #619.

@brandongc FYI

### Todos

- [x] Update the docs

### CODE REVIEW

**Comment by vkarak** [comment]
ok to test

**Comment by teojgo** [comment]
Ok to test

File: reframe/frontend/executors/__init__.py [hunk 1]
```diff
@@ -461,7 +467,11 @@ def on_task_skip(self, task):
 
     @abc.abstractmethod
     def on_task_failure(self, task):
-        '''Called when a regression test has failed.'''
+        '''Called when a RegressionTask has failed.'''
+
+    @abc.abstractmethod
+    def on_task_abort(self, task):
+        '''Called when a RegressionTask has failed.'''
```
Comment (ekouts) [inline reframe/frontend/executors/__init__.py:474]:
```suggestion
        '''Called when a RegressionTask has aborted.'''
```
Maybe you meant to write aborted here?

**Review by ekouts — CHANGES_REQUESTED** [review]
Overall it looks good, if you fix the very small comment I made I will approve.

**Review by ekouts — APPROVED** [review]

**Review by teojgo — APPROVED** [review]

**Pull request merged**